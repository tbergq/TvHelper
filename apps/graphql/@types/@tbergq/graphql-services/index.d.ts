declare module '@tbergq/graphql-services';
