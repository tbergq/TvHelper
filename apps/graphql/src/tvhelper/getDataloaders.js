// @flow

import { FavoritesRepository, type FavoriteType } from '@tbergq/tvhelper-persistence';
import type { LoggedInUser } from '@tbergq/graphql-services';
import Dataloader from 'dataloader';

import SearchTvShowLoader from './search/dataloaders/SearchTvShowLoader';
import UserLoader from './account/dataloaders/UserLoader';
import TvDetailLoader from './tvshow/dataloaders/TvDetailLoader';
import EpisodesLoader from './episode/dataloaders/EpisodesLoader';
import EpisodeLoader from './episode/dataloaders/EpisodeLoader';
import FavoritesLoader from './tvshow/dataloaders/FavoritesLoader';
import EpisodeWatchedLoader from './episode/dataloaders/EpisodeWatched';
import type { TvShow } from './tvshow/TvShow';
import type { Episode, EpisodeWatched } from './episode/Episode';
import type { User } from './account/Account';

export type TvHelperDataLoaders = {
  +searchTvShow: Dataloader<string, TvShow[]>,
  +user: Dataloader<string, ?User>,
  +tvDetail: Dataloader<string, TvShow>,
  +episodes: Dataloader<string, Episode[]>,
  +episode: Dataloader<string, Episode>,
  +favorites: Dataloader<string, FavoriteType[]>,
  +episodeWatched: Dataloader<number, EpisodeWatched>,
};

export default function getDataloaders(user: ?LoggedInUser): TvHelperDataLoaders {
  const favoritesRepository = new FavoritesRepository();
  return {
    searchTvShow: SearchTvShowLoader(),
    user: UserLoader(),
    tvDetail: TvDetailLoader(),
    episodes: EpisodesLoader(),
    episode: EpisodeLoader(),
    favorites: FavoritesLoader(favoritesRepository),
    episodeWatched: EpisodeWatchedLoader(user),
  };
}
