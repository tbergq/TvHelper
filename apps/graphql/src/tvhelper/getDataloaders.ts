import Dataloader from 'dataloader';

import SearchTvShowLoader from './search/dataloaders/SearchTvShowLoader';
import TvDetailLoader from './tvshow/dataloaders/TvDetailLoader';
import EpisodesLoader from './episode/dataloaders/EpisodesLoader';
import EpisodeLoader from './episode/dataloaders/EpisodeLoader';
import EpisodeWatchedLoader from './episode/dataloaders/EpisodeWatched';
import type { TvShow } from './tvshow/TvShow';
import type { Episode, EpisodeWatched } from './episode/Episode';

export type TvHelperDataLoaders = {
  readonly searchTvShow: Dataloader<string, TvShow[]>;
  readonly tvDetail: Dataloader<string, TvShow>;
  readonly episodes: Dataloader<string, Episode[]>;
  readonly episode: Dataloader<string, Episode>;
  readonly episodeWatched: Dataloader<number, EpisodeWatched>;
};
export default function getDataloaders(user: any | null | undefined): TvHelperDataLoaders {
  return {
    searchTvShow: SearchTvShowLoader(),
    tvDetail: TvDetailLoader(),
    episodes: EpisodesLoader(),
    episode: EpisodeLoader(),
    episodeWatched: EpisodeWatchedLoader(user),
  };
}
