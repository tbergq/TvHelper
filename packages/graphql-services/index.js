// @flow

export { default as fetch } from './src/fetch';

// graphql shared types
export { default as LoginType } from './src/shared/types/output/LoginType';
export { default as RangeDelete } from './src/shared/types/output/RangeDelete';

// graphql shared resolvers
export { default as LoginResolver, signToken } from './src/shared/resolvers/LoginResolver';

export type { LoggedInUser } from './src/shared/resolvers/LoginResolver';
export type { Apps } from './src/shared/resolvers/LoginResolver';

// others
export { default as connectionFromArray } from './src/connectionFromArray';
export type { ConnectionArguments, Connection } from './src/connectionFromArray';
export type { GraphQLConnectionDefinitions } from './src/types';
