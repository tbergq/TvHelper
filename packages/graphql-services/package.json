{
  "name": "@tbergq/graphql-services",
  "version": "0.0.1",
  "private": true,
  "main": "index.js",
  "dependencies": {
    "@adeira/fetch": "^1.0.0",
    "@kiwicom/graphql-global-id": "^0.13.0",
    "dataloader": "^1.4.0",
    "dotenv": "^8.2.0",
    "graphql": "^14.5.8",
    "graphql-relay": "^0.6.0",
    "jsonwebtoken": "^8.5.1",
    "password-hash": "^1.2.2"
  }
}
