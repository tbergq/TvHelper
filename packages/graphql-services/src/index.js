// @flow

export { default as fetch } from './fetch';

// graphql shared types
export { default as LoginType } from './shared/types/output/LoginType';
export { default as RangeDelete } from './shared/types/output/RangeDelete';

// graphql shared resolvers
export { default as LoginResolver, signToken } from './shared/resolvers/LoginResolver';

export type { LoggedInUser } from './shared/resolvers/LoginResolver';
export type { Apps } from './shared/resolvers/LoginResolver';

// others
export { default as connectionFromArray } from './connectionFromArray';
export type { ConnectionArguments, Connection } from './connectionFromArray';
export type { GraphQLConnectionDefinitions } from './types';
