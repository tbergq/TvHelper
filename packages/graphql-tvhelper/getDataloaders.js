// @flow

import { FavoritesRepository } from '@tbergq/tvhelper-persistence';

import SearchTvShowLoader from './src/search/dataloaders/SearchTvShowLoader';
import UserLoader from './src/account/dataloaders/UserLoader';
import TvDetailLoader from './src/tvshow/dataloaders/TvDetailLoader';
import EpisodesLoader from './src/episode/dataloaders/EpisodesLoader';
import FavoritesLoader from './src/tvshow/dataloaders/FavoritesLoader';

export default function getDataloaders() {
  const favoritesRepository = new FavoritesRepository();
  return {
    searchTvShow: SearchTvShowLoader(),
    user: UserLoader(),
    tvDetail: TvDetailLoader(),
    episodes: EpisodesLoader(),
    favorites: FavoritesLoader(favoritesRepository),
  };
}
