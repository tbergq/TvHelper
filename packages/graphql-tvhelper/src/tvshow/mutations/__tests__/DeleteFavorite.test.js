// @flow

import { generateExecuteTestQuery } from '@tbergq/graphql-services';
import {
  tvHelperConnection,
  UserRepository,
  FavoritesRepository,
} from '@tbergq/tvhelper-persistence';

import Mutations from '../../../../TvHelperMutations';
import getDataloaders from '../../../../getDataloaders';

let createdUserId;
const context = {
  user: { id: '123' },
  dataLoader: {
    tvhelper: getDataloaders(),
  },
};

const executeTestQuery = generateExecuteTestQuery(null, Mutations, context);

describe('DeleteFavorite', () => {
  beforeEach(async () => {
    const user = await UserRepository.createUser({
      username: 'lol',
      password: 'lol',
      email: 'lol@lol.no',
    });
    createdUserId = user.id;
    await FavoritesRepository.createFavorite(createdUserId, '6');
  });

  afterEach(async () => {
    await tvHelperConnection.collection('users').drop();
    await tvHelperConnection.collection('favorites').drop();
  });

  it('returns success false for wrong user', async () => {
    context.user.id = 'fail';
    const res = await await executeTestQuery(`mutation deleteFavorite {
      deleteFavorite(serieId: "dHZzaG93OjY=") {
        success
      }
    }`);
    expect(res.data.deleteFavorite.success).toBe(false);
  });

  it('returns success false for wrong serieId', async () => {
    context.user.id = createdUserId;
    const res = await await executeTestQuery(`mutation deleteFavorite {
      deleteFavorite(serieId: "dHZzaG93OjY2") {
        success
      }
    }`);
    expect(res.data.deleteFavorite.success).toBe(false);
  });

  it('works with existing favorite', async () => {
    context.user.id = createdUserId;
    const res = await executeTestQuery(`mutation deleteFavorite {
      deleteFavorite(serieId: "dHZzaG93OjY=") {
        success
        id
      }
    }`);
    expect(res).toMatchSnapshot();
  });
});
