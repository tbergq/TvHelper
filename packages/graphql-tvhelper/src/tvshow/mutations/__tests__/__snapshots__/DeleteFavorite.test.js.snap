// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteFavorite works with existing favorite 1`] = `
Object {
  "data": Object {
    "deleteFavorite": Object {
      "id": "dHZzaG93OjY=",
      "success": true,
    },
  },
}
`;
